### ISSUE TITLE
"Tab" isn't perhaps the best spec language?

### ISSUE THREAD

**Issue by slightlyoff** [WICG/prefer-current-tab#4]
While I understand the goal of using `tab` in the name of the API vs. `document`, it might be worth considering some more generalised alternatives that can work in non-tab scenarios (e.g. PWAs). A few that come to mind:

 - `preferCurrentContext`
 - `preferCurrentWindow`
 - `preferCurrentFrame` (perhaps most accurate)

Thoughts?

**Comment by eladalon1983** [comment]
Agreed that `tab` has some problems. The issue has come up a few times, but a better alternative has not yet been found. I'm happy to continue the search and try to find something everybody is happy with.

* `preferCurrentWindow` - I'm concerned about the ambiguity - unclear to Web-developers whether it refers to [window](https://developer.mozilla.org/en-US/docs/Web/API/Window) or to the browser window in which the tab is loaded.
* `preferCurrentFrame` - I agree with your assessment of the accuracy. I'm concerned it would become even more misleading when/if additional APIs are introduced, some of which are already under discussion (`iframe.getViewportMedia` and `element-level capture`).
* `preferCurrentContext` - might be unclear how narrowly/widely "context" is meant - the [browsing context](https://developer.mozilla.org/en-US/docs/Glossary/Browsing_context) or the current iframe. Or wdyt?
* `preferCurrentBrowsingContext` - my own attempt to clarify your suggestion of `preferCurrentContext`. But the previous API shape, `getCurrentBrowsingContextMedia()`, raised a lot of complaints over the length of the name, so we should probably avoid this.

I think `preferCurrentTab` is the lesser evil here. The name has passed muster for @foolip, @alvestrand, @guidou and @yoavweiss. Wdyt? Can you think of something better that could reach similar consensus? If so, I'd be happy to adopt.

**Comment by eladalon1983** [comment]
I've also consulted @alvestrand today and my understanding is that he'd be OK with either `preferCurrentTab` or `preferCurrentBrowsingContext`. Unless you feel strongly against "tab," I'd like to proceed with this name (preferCurrentTab).

**Comment by eladalon1983** [comment]
@slightlyoff, I'm thinking of closing this issue. This knob shipped in m94 and the benefits of renaming do not seem to merit the work it would take. (Also, we don't yet have a name everyone likes better than `preferCurrentTab`.) Wdyt?
